src/app/(sidebar)/smart-contracts/contract-explorer/components/SourceCode.tsx

```diff
@@ -65,7 +72,58 @@ export const SourceCode = ({
     );
   }
 
+  type ContainerResource = "devcontainer" | "codeanywhere" | "github";
+
+  const handleOpenInContainer = (resource: ContainerResource) => {
+    if (!resource) {
+      return;
+    }
+
+    let url = "";
+
+    switch (resource) {
+      case "devcontainer":
+        url = `https://github.com/codespaces/new/${repo}/tree/${commit}`;
+        break;
+      case "codeanywhere":
+        url = `https://app.codeanywhere.com/#https://github.com/${repo}/commit/${commit}`;
+        break;
+      case "github":
+        url = `https://github.com/${repo}/tree/${commit}`;
+        break;
+      default:
+      // Do nothing
+    }
+
+    if (!url) {
+      return;
+    }
+
+    window.open(url, "_blank");
+  };
+
   return (
-    <CodeEditor title="README.md" value={readmeText} selectedLanguage="text" />
+    <CodeEditor
+      title="README.md"
+      value={readmeText}
+      selectedLanguage="text"
+      customEl={
+        <Select
+          id="source-code-viewer"
+          fieldSize="sm"
+          onChange={(e) => {
+            handleOpenInContainer(e.target.value as ContainerResource);
+          }}
+          value=""
+        >
+          <option value="" disabled={true}>
+            Open in
+          </option>
+          <option value="devcontainer">Dev Container</option>
```

Comment:
So just FYI devcontainers is the OSS technology and then Github Codespaces and Codeanywhere are the providers :)

Reply:
We will add more info later :)